### PR TITLE
Issue 12/22. Logging and application error code.

### PR DESCRIPTION
**Pull request by Ramizjon**
Added logging to main places, added error code throwing in case of unsuccessfull application finish

### CODE REVIEW

**Comment by Ramizjon** [comment]
@btrofimov issues 13 and 22. This pull request has conflicts that must be resolved.

**Comment by Ramizjon** [comment]
@btrofimov  updated pull request

File: src/main/java/com/segmentreader/mapreduce/AbstractUserSegmentsMapper.java [hunk 1]
```diff
@@ -14,33 +14,42 @@
...
+        }
+    }
+
+    @Override
+    protected void cleanup(Context context) throws IOException,
+            InterruptedException {
+        for (Closeable closeable : closeables) {
+            closeable.close();
+        }
+        logger.info("Clean up completed");
```
Comment (btrofimov) [inline src/main/java/com/segmentreader/mapreduce/AbstractUserSegmentsMapper.java]: Please change it to debug level

File: src/main/java/com/segmentreader/mapreduce/AppContext.java [hunk 2]
```diff
@@ -13,7 +14,18 @@
 
 public class AppContext {
```
Comment (btrofimov) [inline src/main/java/com/segmentreader/mapreduce/AppContext.java:21]: Please comment the purpose of this class and all inner classes as well in javadoc format.

**Comment by btrofimov** [comment]
please fix  issues above and merge it.

**Pull request merged**